harper-core/src/spell/full_dictionary.rs

```diff
@@ -172,18 +167,101 @@ impl Dictionary for FullDictionary {
         let chars: CharString = word.chars().collect();
         self.get_word_metadata(&chars)
     }
+
+    /// Suggest a correct spelling for a given misspelled word.
+    /// [`word`] is assumed to be quite small (n < 100).
+    /// [`max_distance`] relates to an optimization that allows the search
+    /// algorithm to prune large portions of the search.
+    fn fuzzy_match(
+        &self,
+        word: &[char],
+        max_distance: u8,
+        max_results: usize,
+    ) -> Vec<(&[char], u8, WordMetadata)> {
+        let misspelled_charslice = seq_to_normalized(word);
+
+        let shortest_word_len =
+            std::cmp::max(1, misspelled_charslice.len() - max_distance as usize);
```

Comment:
This modification to the algorithm results in integer underflow for short words.